lib/ts/controllers/s-modal.ts

```diff
@@ -0,0 +1,88 @@
+class ModalController extends Stacks.StacksController {
+    static targets = ["modal"];
+
+    // TODO should Stimulus bind this? Needed to keep TS error-free
+    private modalTarget!: HTMLElement;
+
+    private _boundClickFn!: any;
+    private _boundKeypressFn!: any;
+
+    connect () { };
+
+    toggle (e: Event) {
+        e.preventDefault();
+        this._toggle();
+    }
+
+    _toggle (show?: boolean | undefined) {
+        var toShow = show;
+
+        // if we're letting the class toggle, we need to figure out if the popover is visible manually
+        if (typeof toShow === "undefined") {
+            toShow = this.modalTarget.getAttribute("aria-hidden") === "true";
+        }
+
+        // show/hide events trigger before toggling the class
+        this.triggerEvent(toShow ? "show" : "hide");
+
+        this.modalTarget.setAttribute("aria-hidden", toShow ? "false" : "true");
+        this.modalTarget.setAttribute("tab-index", toShow ? "0" : "-1");
```

Comment:
Several things here. `tab-index` should be `tabindex` but... nothing this code does is really what you want it to do:

- `<div class="s-modal" tabindex="-1"><button></button></div>` does not stop the child button from receiving focus on tab.
- `<div class="s-modal" tabindex="0">` adds a useless focus stop on the entire modal element.

Is there some specific guidance somewhere that is telling us to use `tabindex="-1"` to hide the container from screenreaders?  Otherwise, the combination of `visibility:hidden` and `aria-hidden="true"` are sufficient for assistive technology and keyboard users.